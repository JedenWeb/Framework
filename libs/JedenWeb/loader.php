<?php

/**
 * This file is part of the www.jedenweb.cz webpage (http://www.jedenweb.cz/)
 *
 * Copyright (c) 2012 Pavel Jurásek (jurasekpavel@ctyrimedia.cz), Vojtěch Jurásek (jurasek@ctyrimedia.cz)
 *
 * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
 */

namespace JedenWeb;

use Nette\Diagnostics\Debugger;
use Nette\Application\Routers\SimpleRouter;
use Nette\Application\Routers\Route;
use Nette\Config\Adapters\NeonAdapter;

if (!defined('NETTE')) {
	die('register nette');
	$nettePaths = array(
		__DIR__ . "/../../../../paveljurasek/nette/Nette/loader.php",
		__DIR__ . "/../../../../vendor/paveljurasek/nette/Nette/loader.php",
		__DIR__ . "/../../../../nette/nette/Nette/loader.php",
		__DIR__ . "/../../../../vendor/nette/nette/Nette/loader.php",
	);
	foreach($nettePaths as $path){
		if(file_exists($path)){
			$nettePath = $path;
			break;
		}
	}
	if(!$nettePath){
		die('You must load Nette Framework first');
	}

	include_once $nettePath;
}

define('JEDENWEB', TRUE);
define('JEDENWEB_DIR', __DIR__);
define('JEDENWEB_VERSION', '1.0a');

require_once __DIR__ . '/Config/Configurator.php';