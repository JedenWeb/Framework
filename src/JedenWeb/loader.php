<?php

require_once __DIR__ . '/common/exceptions.php';
