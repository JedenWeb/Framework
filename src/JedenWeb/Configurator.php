<?php

namespace JedenWeb;

use JedenWeb;
use Nette;
use Nette\Application\Routers\Route;

/**
 * @author Pavel JurÃ¡sek <jurasekpavel@ctyrimedia.cz>
 */
class Configurator extends Nette\Configurator
{

	const CACHE_NAMESPACE = 'Nette.Configurator';

	/** @var \Nette\DI\Container */
	protected $container;

	/** @var \Nette\Loaders\RobotLoader */
	protected $robotLoader;

	/** @var Compiler */
	protected $compiler;


	/**
	 * @param array
	 */
	public function __construct($params = array())
	{
		$this->parameters = $this->getDefaultParameters($params);
		$this->setTempDirectory($this->parameters['tempDir']);
	}


	/**
	 * @param array
	 * @return array
	 */
	protected function getDefaultParameters($parameters = array())
	{
		$defaults = parent::getDefaultParameters();

		$defaults['rootDir'] = realpath($defaults['wwwDir'] . '/..');
		$defaults['appDir'] = realpath($defaults['rootDir'] . '/app');
		$defaults['vendorDir'] = realpath($defaults['rootDir'] . '/vendor');
		$defaults['tempDir'] = realpath($defaults['rootDir'] . '/temp');
		$defaults['logDir'] = realpath($defaults['rootDir'] . '/log');
		$defaults['configDir'] = realpath($defaults['appDir'] . '/config');

		return array_merge($defaults, $parameters);
	}


	/**
	 * @return \Nette\DI\Container
	 */
	public function getContainer()
	{
		if (!$this->container) {
			$this->container = $this->createContainer();
		}

		return $this->container;
	}


	/**
	 * @return \Nette\DI\Container
	 */
	public function createContainer()
	{
		foreach ($this->getConfigFiles() as $file) {
			if (is_file($file)) {
				$this->addConfig($file, self::NONE);
			}
		}

		$container = parent::createContainer();
		$container->addService('configurator', $this);

		$container->getService('router')
			->offsetSet(NULL, new Route('index.php', 'Homepage:default', Route::ONE_WAY));

		return $container;
	}


	/**
	 * @return \Nette\DI\Compiler
	 */
	protected function createCompiler()
	{
		$this->compiler = parent::createCompiler();
		$this->compiler
				->addExtension('jedenWeb', new DI\Extensions\JedenWebExtension);

		return $this->compiler;
	}


	/**
	 * @return array
	 */
	protected function getConfigFiles()
	{
		$configs = array(
			$this->parameters['configDir'] . "/config.neon",
			$this->parameters['configDir'] . "/config.local.neon",
		);

		return $configs;
	}


	public function enableRobotLoader()
	{
		$this->robotLoader = $this->createRobotLoader();
		$this->robotLoader
				->addDirectory($this->parameters['appDir'])
				->register();
	}


	/**
	 * @param string
	 * @param string
	 */
	public function enableDebugger($logDirectory = NULL, $email = NULL)
	{
		$logDirectory = $logDirectory ?: $this->parameters['logDir'];

		parent::enableDebugger($logDirectory, $email);
	}


	/**
	 * @return \Nette\DI\Compiler
	 */
	public function getCompiler()
	{
		return $this->compiler;
	}

}
