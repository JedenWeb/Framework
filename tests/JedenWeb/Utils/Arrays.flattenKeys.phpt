<?php

/**
 * Test: JedenWeb\Utils\Arrays
 *
 * @author     Pavel Jurasek
 * @package    JedenWeb\Utils
 */

use JedenWeb\Utils\Arrays;


require __DIR__ . '/../bootstrap.php';

//

$input = array(
	'a' => 1,
	'b' => 2,
	array(
		'c' => 3,
		'd' => 4
	),
);
$output = array('a' => 1, 'b' => 2, 'c' => 3, 'd' => 4);

\Tester\Assert::same($output, Arrays::flattenKeys($input));

//

$input = array(
	1,
	array(2)
);

\Tester\Assert::exception(function() use ($input) {
	Arrays::flattenKeys($input);
}, 'JedenWeb\InvalidStateException');

//

$input = array(
	1 => 'a',
	2 => 'b',
	array('c', 'd', 'e'),
);
$output = array(1 => 'a', 2 => 'b', 0 => 'c', 3 => 'd', 4 => 'e');

\Tester\Assert::same($output, Arrays::flattenKeys($input, FALSE));
