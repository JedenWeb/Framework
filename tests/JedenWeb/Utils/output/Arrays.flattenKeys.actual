array(1 => 'a', 'b', 0 => 'c', 'd', 'e')