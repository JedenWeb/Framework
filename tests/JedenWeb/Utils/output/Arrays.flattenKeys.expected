array('a', 'b', 'c', 'd', 'e')